File: 242-valid-anagram/242-valid-anagram.scala
object Solution {
    def isAnagram(s: String, t: String): Boolean = {
        val cnt = Array.fill(26) {0}
        
        s.foreach { x =>
            val idx: Int = x - 'a'
            cnt(idx) += 1
        }
        
        t.foreach { x =>
            val idx: Int = x - 'a'
            cnt(idx) -= 1
        }
        
        cnt.map(_ == 0).reduce(_ && _)
    }
}

File: 242-valid-anagram/242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] cnt = new int[26];
        
        if (s.length() != t.length()) return false;
        
        for(int i = 0; i < s.length(); i++) {
            cnt[(int)(s.charAt(i) - 'a')] += 1;
            cnt[(int)(t.charAt(i) - 'a')] -= 1;
        }
        
        for(int i = 0; i < cnt.length; i++) {
            if (cnt[i] != 0) return false;
        }
        
        return true;
    }
}